plugins/modules/aws_s3.py

```diff
@@ -634,6 +667,46 @@ def get_download_url(module, s3, bucket, obj, expiry, changed=True):
         module.fail_json_aws(e, msg="Failed while getting download url.")
 
 
+def copy_object_to_bucket(module, s3, bucket, obj, encrypt, metadata, validate, d_etag):
```

Comment:
I would prefer this PR https://github.com/ansible-collections/amazon.aws/pull/335/files to be merged first, if you agree @jillr. I think copying an object from one bucket to another should preserve some properties of the object such as the storage class, tags if the user doesn't explicitly specify different ones for the copied object.

Reply:
335 has been merged, this PR will need to be rebased/have the conflicts resolved.  I agree we should preserve object properties when copying it if the user doesn't specify new properties.